### DISCUSSION TITLE
is it possible to point to a local certificate?

### DISCUSSION THREAD

**Discussion by PyDataBlog** [olimorris/codecompanion.nvim#498] [Q&A] [answered]
can't use this on my work PC as it needs access via a local .pem certificate. currently curl can't point to it when it tries to connect to the GitHub API via curl. How can I add my certificate to the plugin?

**Comment by olimorris** [comment]
Digging into https://github.com/nvim-lua/plenary.nvim/blob/master/lua/plenary/curl.lua, it doesn't look like certificates are supported.

How does curl handle them outside of plenary?

**Reply by PyDataBlog** [reply]
> Digging into https://github.com/nvim-lua/plenary.nvim/blob/master/lua/plenary/curl.lua, it doesn't look like certificates are supported.
> 
> How does curl handle them outside of plenary?

curl --cert certificate.pem https://reqbin.com/echo

**Reply by olimorris** [reply, marked as answer]
In which case, I think you can update your chosen adapter's `raw` table to include `--cert certificate.pem`. So something like:

```lua
adapters = {
  anthropic = function()
    return require("codecompanion.adapters").extend("anthropic", {
      raw = {
        "--cert certificate.pem",
        "--no-buffer",
        "--silent",
      }
    })
  end,
}
```

**Reply by olimorris** [reply]
@PyDataBlog were you able to get this to work?

**Reply by PyDataBlog** [reply]
> @PyDataBlog were you able to get this to work?

will try it tomorrow and report back :)
